app/models/concerns/module_settings_concern.rb

```diff
@@ -0,0 +1,35 @@
+module ModuleSettingsConcern
```

Comment:
I guess that one would be another concern ? Cause this concern is for both instance and course, for them to disable/enable modules.